packages/instantsearch.js/src/lib/utils/hydrateRecommendCache.ts

```diff
@@ -0,0 +1,19 @@
+import type { InitialResults } from '../../types';
+import type { AlgoliaSearchHelper } from 'algoliasearch-helper';
+
+export function hydrateRecommendCache(
+  helper: AlgoliaSearchHelper,
+  initialResults: InitialResults
+) {
+  const recommendCache = Object.keys(initialResults).reduce(
+    (acc, indexName) => {
+      const initialResult = initialResults[indexName];
+      if (initialResult.recommendResults) {
+        return { ...acc, ...initialResult.recommendResults.results };
```

Comment:
let's loop without spreading acc maybe? not sure it's easy as we want to merge though. The equivalent I may be thinking of is `Object.assign({}, ...Object.keys(initialResults).map(key => initialResults[indexName].recommendResults.results))`, possibly worth benchmarking if it's worth the difference though as that may not be super efficient either. We can also use the `merge` function (is it also here or in the helper only) instead of object.assign possibly. At least that would avoid the constant iteration over the acc

Reply:
Good point, and I did do a benchmark : https://jsbench.me/oulwm3j29m/1
It's not as big a difference I thought would be on Chromium surprisingly (it is on WebKit though I guess ?), but still it's better yeah :D

Reply:
We don't allow `Object.assign` in babel config 🙃